### PR TITLE
TiFlash: Fix playground stability

### LINKED ISSUE #233: playground is not stable
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?

```
tiup --tag=super playground
tiup clean --all
tiup --tag=super playground
```

2. What did you expect to see?
should run normally 

3. What did you see instead?
TiFlash down after second playground execution.
```
Starting component `playground`: /root/.tiup/components/playground/v0.0.9/playground
Playground Bootstrapping...
Starting component `pd`: /root/.tiup/components/pd/v4.0.0-rc.2/pd-server --name=pd-0 --data-dir=/root/.tiup/data/RzW5KxQ/pd-0/data --peer-urls=http://127.0.0.1:2380 --advertise-peer-urls=http://127.0.0.1:2380 --client-urls=http://127.0.0.1:2379 --advertise-client-urls=http://127.0.0.1:2379 --log-file=/root/.tiup/data/RzW5KxQ/pd-0/pd.log --config=/root/.tiup/data/RzW5KxQ/pd-0/pd.toml --initial-cluster=pd-0=http://127.0.0.1:2380
Starting component `tikv`: /root/.tiup/components/tikv/v4.0.0-rc.2/tikv-server --addr=127.0.0.1:20160 --status-addr=127.0.0.1:33292 --pd=http://127.0.0.1:2379 --config=/root/.tiup/data/RzW5KxQ/tikv-0/tikv.toml --data-dir=/root/.tiup/data/RzW5KxQ/tikv-0/data --log-file=/root/.tiup/data/RzW5KxQ/tikv-0/tikv.log
Starting component `tidb`: /root/.tiup/components/tidb/v4.0.0-rc.2/tidb-server -P 4000 --store=tikv --host=127.0.0.1 --status=10080 --path=127.0.0.1:2379 --log-file=/root/.tiup/data/RzW5KxQ/tidb-0/tidb.log
CLUSTER START SUCCESSFULLY, Enjoy it ^-^
To connect TiDB: mysql --host 127.0.0.1 --port 4000 -u root
To view the dashboard: http://127.0.0.1:2379/dashboard
Starting component `tiflash`: /root/.tiup/components/tiflash/v4.0.0-rc.2/tiflash/tiflash server --config-file=/root/.tiup/data/RzW5KxQ/tiflash-0/tiflash.toml
Logging debug to /root/.tiup/data/RzW5KxQ/tiflash-0/log/tiflash.log
Logging errors to /root/.tiup/data/RzW5KxQ/tiflash-0/log/tiflash_error.log
Error: run `/root/.tiup/components/tiflash/v4.0.0-rc.2/tiflash/tiflash` (wd:/root/.tiup/data/RzW5KxQ/tiflash-0) failed: exit status 70
```

4. What version of TiUP are you using (`tiup --version`)?
v0.0.9

### PR DESCRIPTION
**Pull request by birdstorm**
### What problem does this PR solve? 
Close #233 

### What is changed and how it works?
Add more check for TiFlash so that TiFlash is determined to start.

### Check List 

Tests 

 - Manual test 

Code changes

 - Has exported function/method change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: components/playground/main.go [hunk 1]
```diff
@@ -411,10 +430,45 @@ func bootCluster(options *bootOptions) error {
...
+		for _, kv := range kvs {
+			if err := checkStoreStatus(pdClient, kv.StoreAddr()); err != nil {
+				lastErr = err
+				break
+			}
+		}
+
+		for _, flash := range flashs {
+			if err := flash.Start(ctx, v0manifest.Version(options.version), pathMap["tiflash"], profile); err != nil {
+				lastErr = err
```
Comment (lichunzhu) [inline components/playground/main.go:452]: shall we break here?

File: components/playground/main.go [hunk 2]
```diff
@@ -411,10 +430,45 @@ func bootCluster(options *bootOptions) error {
...
+			}
+		}
+
+		// check if all TiFlash is up
+		for _, flash := range flashs {
+			if err := checkStoreStatus(pdClient, flash.StoreAddr()); err != nil {
+				lastErr = err
+				break
+			}
+		}
```
Comment (lichunzhu) [inline components/playground/main.go:462]: if `lastErr` is not nil, should we skip following for loops?

File: components/playground/utils/utils.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	} else {
+		cfg = RetryOption{
+			Attempts: defaultAttempts,
+			Delay:    defaultDelay,
+			Timeout:  defaultTimeout,
+		}
+	}
+
+	// timeout must be greater than 0
+	if cfg.Timeout <= 0 {
```
Comment (lichunzhu) [inline components/playground/utils/utils.go:50]: How about setting options automatically for invalid value like `cfg. Delay`?

**Comment by lichunzhu** [comment]
@birdstorm please resolve the conflicts

File: components/playground/utils/utils.go (hunk 3 above)
Comment (birdstorm) [reply components/playground/utils/utils.go:50]: Just copied these lines from tiup-cluster

**Pull request merged**